Add dogeos bridge

shu-unifra opened the pull request.

shu-unifra committed:
support some dogeos config values

shu-unifra committed:
update image tag

shu-unifra committed:
auto generate yaml file for frontends

shu-unifra committed:
dogeos-values=>values

shu-unifra committed:
ingress process for celesitia and tso

shu-unifra committed:
some fix

shu-unifra committed:
some missing values

shu-unifra committed:
more fix

shu-unifra committed:
add default value for testnet

shu-unifra committed:
dogeos69 docker hub

shu-unifra committed:
cs command auto login and create session

shu-unifra committed:
move cubesigner to a standalone file

shu-unifra committed:
fix metrics config generation bug

shu-unifra committed:
some dogecoin configs

shu-unifra committed:
1. dogecoin ingress

2. some fix

shu-unifra committed:
dogeos tls

shu-unifra committed:
turn DOGEOS_DEPOSIT_PROCESSOR_NOWNODES_API_KEY into a secret

shu-unifra committed:
update README.md

shu-unifra committed:
fix metrics L1_MESSAGE_QUEUE_V2_PROXY_ADDR

shu-unifra committed:
fix dogecoin service name

shu-unifra committed:
add secret DOGEOS_CUBESIGNER_SIGNER_CS_KEY_ID

shu-unifra committed:
fix CELESTIA_URL

dghelm commented on src/commands/setup/cubesigner.ts:
```diff
@@ -0,0 +1,118 @@
+            const loginSuccess = await this.executeInteractiveCommand('cs', [
+                'login',
+                '--env',
+                'gamma',
```
This should be a parameter we can set since this is not the production environment for CubeSigner.

shu-unifra replied:
so, make gamma a parameter

dghelm commented on src/commands/setup/domains.ts:
```diff
@@ -327,6 +327,9 @@ export default class SetupDomains extends Command {
         RPC_GATEWAY_HOST: `l2-rpc.${urlEnding}`,
         ...(usesAnvil ? {L1_DEVNET_HOST: `l1-devnet.${urlEnding}`, L1_EXPLORER_HOST: `l1-explorer.${urlEnding}`} : {}),
+        TSO_HOST: `tso.${urlEnding}`,
+        CELESTIA_HOST: `celestia.${urlEnding}`,
```
I don't believe Celestia or Dogecoin should have a public-facing ingress. @alexanvl to confirm.

dghelm commented on src/config/setup_defaults.toml:
```diff
@@ -0,0 +1,68 @@
+# --- Dogecoin RPC --- 
+dogecoin_rpc_url = "https://testnet.doge.xyz"
+dogecoin_rpc_user = "user"
+dogecoin_rpc_pass = "password_test"
```
Probably not great that we're added this to a public repo. We should consider additional protections or changing the password.

shu-unifra replied:
yes.

dghelm commented on src/config/setup_defaults.toml:
```diff
@@ -0,0 +1,68 @@
+# Optional: Blockbook URL for faster UTXO fetching (highly recommended)
+# If not provided, UTXO fetching might be slower or require manual input.
+dogecoin_blockbook_url = "https://dogebook-testnet.nownodes.io"
+dogecoin_blockbook_api_key = "7ef0af1d-43c2-45e5-9b96-76421a9f2f95" # If required
```
Public repo -- @alexanvl we need to burn this key and maybe genereate a few new nownodes ones, for test and production.

shu-unifra replied:
Yes

dghelm commented on src/config/setup_defaults.toml:
```diff
@@ -0,0 +1,68 @@
+# --- Base Identity --- 
+# Seed string used to derive the base private key.
+# Recommend using environment variable TESTNET_SETUP_SEED_STRING instead.
+seed_string = "initial_tests-eeeejjj-05"
```
We need a way to make sure the user does not use this at its default value (or even a predictable one) for any public testnet. Find for internal devnets, but knowing this equivalent to knowing the private key of sequencer and fee wallet (unless pubkeys are manually provided elsewhere in this file, which needs to be the flow for any public testnet).

shu-unifra replied:
ok, will remove the default value.

dghelm commented on src/utils/doge-config.ts:
```diff
@@ -38,25 +38,36 @@ export async function loadDogeConfig(configPath: string): Promise<DogeConfig> {
+        celestiaIndexerStartBlock: network === 'mainnet' ? '0' : '6338800',
+        //rpcUrl: network === 'mainnet' ? 'http://celestia-mainnet:26658' : 'http://celestia-testnet-mocha:26658',
+        tendermintRpcUrl: '',
+        daNamespace: network === 'mainnet' ? '' : '98E6DED48612C0B8E4FA',
```
I'm not sure I understand these default values or why tendermintRPC is empty.

shu-unifra replied:
this value from alex repo is like:
https://small-neat-frog.celestia-mocha.quiknode.pro/xxxxxxxxxxxxxxxxxxx
I think may be it  contains a token, so I did not set a default value here.

shu-unifra committed:
remove some password and privte key

add new command 'doge bridge-init'

some bug fix

shu-unifra committed:
move 'select deploy type' to top

tony-unifra committed:
chore: remove baiscAuth config of metrics-exporter

shu-unifra committed:
setup signer

shu-unifra committed:
update dummy signer remote tag

shu-unifra committed:
some fix

shu-unifra committed:
rename cubesigner => cubesigner-refresh

support push cubesigner secrets only

shu-unifra committed:
fix: resolve dummy-signers WIF generation and improve setup UX

- Use bitcore-lib-doge for WIF generation (fixes "Invalid WIF format" error)
- Add network selection (regtest/testnet/mainnet) for WIF generation
- Improve error messages to guide users through proper setup flow
- Enhance prompts for AWS suffixes and RPC credentials
- Remove display of sensitive credentials from console output

shu-unifra committed:
remove a unused value

shu-unifra committed:
update contracts tag to latest

remove DOGEOS_WITHDRAWAL_CELESTIA_INDEXER__TENDERMINT_RPC_URL from env

shu-unifra committed:
remove unused variable and functio

shu-unifra committed:
feat: use internal K8s service for Dogecoin RPC and standardize YAML output format

- Change withdrawal-processor and metrics-exporter to use internal Dogecoin service URL instead of external ingress (http://dogecoin-{network}:port)
- Add YAML_DUMP_OPTIONS constant in config/constants.ts for consistent formatting
- Update prep-charts, push-secrets, and tls commands to use unified YAML configuration
- Set forceQuotes=true to prevent YAML parsing ambiguities
- Update default config filename from \"doge-config.toml\" to \"doge-config-testnet.toml\"

shu-unifra committed:
tmp

shu-unifra committed:
a command disable internal ingress

shu-unifra committed:
othere default values
